app/lib/git_repository.rb

```diff
@@ -395,7 +395,13 @@ def commit(transaction)
         end
       when :remove
         begin
-          remove_file(job[:path], job[:expected_revision_identifier])
+          remove_file(job[:path], job[:expected_revision_identifier], keep_folder: job[:keep_folder])
+        rescue Repository::Conflict => e
+          transaction.add_conflict(e)
+        end
+      when :remove_directory
```

Comment:
Please add this case to the `commit` method for both the memory and subversion repository as well.